### PR TITLE
Upgrade gradle-errorprone

### PR DESCRIPTION
**Pull request by ferozco**
## Before this PR
We had blocked upgrades of gradle-errorprone because we were worried about forcing a "high" dependency on gradle (5.2). Since then gradle has released a bunch and the fleet has been keeping up to date so we no longer have to be worried about the dependency.

## After this PR
==COMMIT_MSG==
Upgrade gradle-errorprone
==COMMIT_MSG==

## Possible downsides?
N/A

### CODE REVIEW

**Comment by robert3005** [comment]
errorProneArgs is now a property so you can't do https://github.com/palantir/gradle-baseline/blob/develop/gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineErrorProneIntegrationTest.groovy#L173

**Comment by ferozco** [comment]
Ya I noticed :( its quite unfortunate since its pretty common for people to configure checks that way. I think we may want to hold off on merging this until we have a replacement. My current thoughts are to propose the following to the upstream plugin:
```
tasks.withType(JavaCompile) {
  options.errorprone.check 'OptionalOrElseMethodInvocation', 'OFF'
}
```

Eh, actually looking at it they already expose pretty nice methods configuring checks. ex: https://github.com/palantir/gradle-baseline/blob/d039b4b123daac27e8e3083eaeda9b68936eec99/gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineErrorProneIntegrationTest.groovy#L173

**Comment by robert3005** [comment]
yeah, I think current version of the plugin has nice apis. Just the original version only really had the array based method and we stuck to it.

File: README.md [hunk 1]
```diff
@@ -124,11 +124,7 @@ Rules can be suppressed at the project level, or have their severity modified, b
 
 ```gradle
 tasks.withType(JavaCompile).configureEach {
-    options.errorprone.errorproneArgs += ['-Xep:Slf4jLogsafeArgs:OFF']
-    // alternatively, using the DSL:
-    options.errorprone {
-        check('Slf4jLogsafeArgs', CheckSeverity.OFF)
-    }
+    options.errorprone.disable '-Slf4jLogsafeArgs'
```
Comment (robert3005) [inline README.md]: I think you don't want the dash at the beginning?

**Review by robert3005 — APPROVED** [review]

**Pull request merged**